databricks/koalas/frame.py

```diff
@@ -10049,6 +10049,80 @@ def explode(self, column):
         internal = self._internal.copy(spark_frame=sdf, data_spark_columns=data_scols)
         return DataFrame(internal)
 
+    def mad(self, axis=0):
+        """
+        Return the mean absolute deviation of values.
+
+        Parameters
+        ----------
+        axis : {index (0), columns (1)}
+            Axis for the function to be applied on.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'a': [1, 2, 3, np.nan], 'b': [0.1, 0.2, 0.3, np.nan]},
+        ...                   columns=['a', 'b'])
+
+        >>> df.mad()
+        a    0.666667
+        b    0.066667
+        Name: 0, dtype: float64
+
+        >>> df.mad(axis=1)
+        0    0.45
+        1    0.90
+        2    1.35
+        3     NaN
+        Name: 0, dtype: float64
+        """
+        from databricks.koalas import Series
+        from databricks.koalas.series import first_series
+
+        axis = validate_axis(axis)
+
+        if axis == 0:
+            exprs = []
+            for label in self._internal.column_labels:
+                scol = self._internal.spark_column_for(label)
+                col_type = self._internal.spark_type_for(label)
+
+                if isinstance(col_type, BooleanType):
+                    scol = scol.cast("integer")
+
+                label_str = name_like_string(label)
+
+                scol = F.avg(
+                    F.abs(scol - self._sdf.select(F.avg(scol).alias(label_str)).first()[label_str])
```

Comment:
Hmm... I think we'd better to avoid using `first()` for every single loop since It occurs a Spark job.

Reply:
I think you are right. I refactored to call first() only once.